### ISSUE TITLE
Bug: AUTH_FAILED on new PIA, works fine on PIA Old

### ISSUE THREAD

**Issue by dfsimmons7** [qdm12/gluetun#265]
openvpn: AUTH: Received control message: AUTH_FAILED when using PIA.  Same creds work fine on PIA Old.

1. Is this urgent?

    - [ ] Yes
    - [x] No

2. What VPN service provider are you using?

    - [x] PIA
    - [ ] Mullvad
    - [ ] Windscribe
    - [ ] Surfshark
    - [ ] Cyberghost

3. What's the version of the program?

 Running version latest built on 2020-10-20T18:14:45Z (commit bf92008)

4. What are you using to run the container?

    - [ ] Docker run
    - [x] Docker Compose
    - [ ] Kubernetes
    - [ ] Docker stack
    - [ ] Docker swarm
    - [ ] Podman
    - [ ] Other:

5. Extra information

Logs:

```
2020-10-21T16:42:49.760Z        INFO    OpenVPN version: 2.4.9
2020-10-21T16:42:49.798Z        INFO    Unbound version: 1.10.1
2020-10-21T16:42:49.828Z        INFO    IPtables version: v1.8.4
2020-10-21T16:42:49.925Z        INFO    TinyProxy version: 1.10.0
2020-10-21T16:42:49.926Z        INFO    Settings summary below:
OpenVPN settings:
|--User: [redacted]
|--Password: [redacted]
|--Verbosity level: 1
|--Run as root: no
|--Private Internet Access settings:
 |--Network protocol: udp
 |--Regions: us west
 |--Encryption preset: strong
 |--Port forwarding: off
System settings:
|--User ID: 1000
|--Group ID: 1000
|--Timezone:
|--IP Status filepath: /tmp/gluetun/ip
DNS over TLS settings:
 |--DNS over TLS provider:
  |--cloudflare
 |--Caching: enabled
 |--Block malicious: enabled
 |--Block surveillance: disabled
 |--Block ads: disabled
 |--Allowed hostnames:
  |--
 |--Private addresses:
  |--127.0.0.1/8
  |--10.0.0.0/8
  |--172.16.0.0/12
  |--192.168.0.0/16
  |--169.254.0.0/16
  |--::1/128
  |--fc00::/7
  |--fe80::/10
  |--::ffff:0:0/96
 |--Verbosity level: 1/5
 |--Verbosity details level: 0/4
 |--Validation log level: 0/2
 |--IPv6 resolution: disabled
 |--Update: every 24h0m0s
 |--Keep nameserver (disabled blocking): no
Firewall settings:
 |--Allowed subnets: 192.168.0.0/24
 |--VPN input ports:
 |--Input ports:
TinyProxy settings: disabled
ShadowSocks settings: disabled
HTTP Control server:
 |--Listening port: 8000
 |--Logging: true
Public IP check period: 12h0m0s
Version information: enabled
Updater: disabled

2020-10-21T16:42:49.927Z        INFO    storage: Merging by most recent 6398 hardcoded servers and 0 servers read from /gluetun/servers.json
2020-10-21T16:42:49.972Z        INFO    routing: default route found: interface eth0, gateway 172.17.0.1
2020-10-21T16:42:49.972Z        INFO    routing: local subnet found: 172.17.0.0/16
2020-10-21T16:42:49.972Z        INFO    openvpn configurator: checking for device /dev/net/tun
2020-10-21T16:42:49.972Z        WARN    TUN device is not available: open /dev/net/tun: no such file or directory
2020-10-21T16:42:49.972Z        INFO    openvpn configurator: creating /dev/net/tun
2020-10-21T16:42:49.973Z        INFO    firewall: enabling...
2020-10-21T16:42:50.052Z        INFO    firewall: enabled successfully
2020-10-21T16:42:50.052Z        INFO    firewall: setting allowed subnets through firewall...
2020-10-21T16:42:50.054Z        INFO    routing: adding 192.168.0.0/24 as route via 172.17.0.1 eth0
2020-10-21T16:42:50.147Z        INFO    http server: listening on 0.0.0.0:8000
2020-10-21T16:42:50.147Z        INFO    Launching standard output merger
2020-10-21T16:42:50.147Z        INFO    dns over tls: falling back on plaintext DNS at address 1.1.1.1
2020-10-21T16:42:50.147Z        INFO    dns configurator: using DNS address 1.1.1.1 internally
2020-10-21T16:42:50.147Z        INFO    dns configurator: using DNS address 1.1.1.1 system wide
2020-10-21T16:42:50.198Z        INFO    firewall: setting VPN connection through firewall...
2020-10-21T16:42:50.254Z        INFO    openvpn configurator: starting openvpn
2020-10-21T16:42:50.256Z        INFO    openvpn: OpenVPN 2.4.9 x86_64-alpine-linux-musl [SSL (OpenSSL)] [LZO] [LZ4] [EPOLL] [MH/PKTINFO] [AEAD] built on Apr 20 2020
2020-10-21T16:42:50.256Z        INFO    openvpn: library versions: OpenSSL 1.1.1g  21 Apr 2020, LZO 2.10
2020-10-21T16:42:50.296Z        INFO    openvpn: CRL: loaded 1 CRLs from file [[INLINE]]
2020-10-21T16:42:50.297Z        INFO    openvpn: TCP/UDP: Preserving recently used remote address: [AF_INET]184.170.241.67:1197
2020-10-21T16:42:50.297Z        INFO    openvpn: UDP link local: (not bound)
2020-10-21T16:42:50.297Z        INFO    openvpn: UDP link remote: [AF_INET]184.170.241.67:1197
2020-10-21T16:42:51.020Z        WARN    openvpn: 'link-mtu' is used inconsistently, local='link-mtu 1570', remote='link-mtu 1542'
2020-10-21T16:42:51.020Z        WARN    openvpn: 'cipher' is used inconsistently, local='cipher AES-256-CBC', remote='cipher BF-CBC'
2020-10-21T16:42:51.020Z        WARN    openvpn: 'auth' is used inconsistently, local='auth SHA256', remote='auth SHA1'
2020-10-21T16:42:51.020Z        WARN    openvpn: 'keysize' is used inconsistently, local='keysize 256', remote='keysize 128'
2020-10-21T16:42:51.020Z        INFO    openvpn: [phoenix407] Peer Connection Initiated with [AF_INET]184.170.241.67:1197
2020-10-21T16:42:52.546Z        INFO    openvpn: AUTH: Received control message: AUTH_FAILED
2020-10-21T16:42:52.546Z        INFO    openvpn: SIGUSR1[soft,auth-failure] received, process restarting
2020-10-21T16:43:02.547Z        INFO    openvpn: TCP/UDP: Preserving recently used remote address: [AF_INET]184.170.241.67:1197
2020-10-21T16:43:02.548Z        INFO    openvpn: UDP link local: (not bound)
2020-10-21T16:43:02.548Z        INFO    openvpn: UDP link remote: [AF_INET]184.170.241.67:1197
2020-10-21T16:43:03.653Z        WARN    openvpn: 'link-mtu' is used inconsistently, local='link-mtu 1570', remote='link-mtu 1542'
2020-10-21T16:43:03.653Z        WARN    openvpn: 'cipher' is used inconsistently, local='cipher AES-256-CBC', remote='cipher BF-CBC'
2020-10-21T16:43:03.653Z        WARN    openvpn: 'auth' is used inconsistently, local='auth SHA256', remote='auth SHA1'
2020-10-21T16:43:03.653Z        WARN    openvpn: 'keysize' is used inconsistently, local='keysize 256', remote='keysize 128'
2020-10-21T16:43:03.653Z        INFO    openvpn: [phoenix407] Peer Connection Initiated with [AF_INET]184.170.241.67:1197
2020-10-21T16:43:04.763Z        INFO    openvpn: AUTH: Received control message: AUTH_FAILED
2020-10-21T16:43:04.763Z        INFO    openvpn: SIGUSR1[soft,auth-failure] received, process restarting
```

PIA OLD Logs:

```
2020-10-21T16:39:34.307Z        INFO    TinyProxy version: 1.10.0
2020-10-21T16:39:34.442Z        INFO    OpenVPN version: 2.4.9
2020-10-21T16:39:34.487Z        INFO    Unbound version: 1.10.1
2020-10-21T16:39:34.632Z        INFO    IPtables version: v1.8.4
2020-10-21T16:39:34.632Z        INFO    Settings summary below:
OpenVPN settings:
|--User: [redacted]
|--Password: [redacted]
|--Verbosity level: 1
|--Run as root: no
|--Private Internet Access Old settings:
 |--Network protocol: udp
 |--Regions: us west
 |--Encryption preset: strong
 |--Port forwarding: off
System settings:
|--User ID: 1000
|--Group ID: 1000
|--Timezone:
|--IP Status filepath: /tmp/gluetun/ip
DNS over TLS settings:
 |--DNS over TLS provider:
  |--cloudflare
 |--Caching: enabled
 |--Block malicious: enabled
 |--Block surveillance: disabled
 |--Block ads: disabled
 |--Allowed hostnames:
  |--
 |--Private addresses:
  |--127.0.0.1/8
  |--10.0.0.0/8
  |--172.16.0.0/12
  |--192.168.0.0/16
  |--169.254.0.0/16
  |--::1/128
  |--fc00::/7
  |--fe80::/10
  |--::ffff:0:0/96
 |--Verbosity level: 1/5
 |--Verbosity details level: 0/4
 |--Validation log level: 0/2
 |--IPv6 resolution: disabled
 |--Update: every 24h0m0s
 |--Keep nameserver (disabled blocking): no
Firewall settings:
 |--Allowed subnets: 192.168.0.0/24
 |--VPN input ports:
 |--Input ports:
TinyProxy settings: disabled
ShadowSocks settings: disabled
HTTP Control server:
 |--Listening port: 8000
 |--Logging: true
Public IP check period: 12h0m0s
Version information: enabled
Updater: disabled

2020-10-21T16:39:34.635Z        INFO    storage: Merging by most recent 6398 hardcoded servers and 0 servers read from /gluetun/servers.json
2020-10-21T16:39:34.660Z        INFO    routing: default route found: interface eth0, gateway 172.17.0.1
2020-10-21T16:39:34.660Z        INFO    routing: local subnet found: 172.17.0.0/16
2020-10-21T16:39:34.660Z        INFO    openvpn configurator: checking for device /dev/net/tun
2020-10-21T16:39:34.660Z        WARN    TUN device is not available: open /dev/net/tun: no such file or directory
2020-10-21T16:39:34.660Z        INFO    openvpn configurator: creating /dev/net/tun
2020-10-21T16:39:34.661Z        INFO    firewall: enabling...
2020-10-21T16:39:34.687Z        INFO    firewall: enabled successfully
2020-10-21T16:39:34.687Z        INFO    firewall: setting allowed subnets through firewall...
2020-10-21T16:39:34.689Z        INFO    routing: adding 192.168.0.0/24 as route via 172.17.0.1 eth0
2020-10-21T16:39:34.801Z        INFO    http server: listening on 0.0.0.0:8000
2020-10-21T16:39:34.802Z        INFO    dns over tls: falling back on plaintext DNS at address 1.1.1.1
2020-10-21T16:39:34.802Z        INFO    dns configurator: using DNS address 1.1.1.1 internally
2020-10-21T16:39:34.802Z        INFO    dns configurator: using DNS address 1.1.1.1 system wide
2020-10-21T16:39:34.802Z        INFO    Launching standard output merger
2020-10-21T16:39:34.825Z        INFO    firewall: setting VPN connection through firewall...
2020-10-21T16:39:34.875Z        INFO    openvpn configurator: starting openvpn
2020-10-21T16:39:34.878Z        INFO    openvpn: OpenVPN 2.4.9 x86_64-alpine-linux-musl [SSL (OpenSSL)] [LZO] [LZ4] [EPOLL] [MH/PKTINFO] [AEAD] built on Apr 20 2020
2020-10-21T16:39:34.878Z        INFO    openvpn: library versions: OpenSSL 1.1.1g  21 Apr 2020, LZO 2.10
2020-10-21T16:39:34.910Z        INFO    openvpn: CRL: loaded 1 CRLs from file [[INLINE]]
2020-10-21T16:39:34.910Z        INFO    openvpn: TCP/UDP: Preserving recently used remote address: [AF_INET]184.170.242.199:1197
2020-10-21T16:39:34.910Z        INFO    openvpn: UDP link local: (not bound)
2020-10-21T16:39:34.910Z        INFO    openvpn: UDP link remote: [AF_INET]184.170.242.199:1197
2020-10-21T16:39:37.399Z        WARN    openvpn: 'link-mtu' is used inconsistently, local='link-mtu 1570', remote='link-mtu 1542'
2020-10-21T16:39:37.399Z        WARN    openvpn: 'cipher' is used inconsistently, local='cipher AES-256-CBC', remote='cipher BF-CBC'
2020-10-21T16:39:37.399Z        WARN    openvpn: 'auth' is used inconsistently, local='auth SHA256', remote='auth SHA1'
2020-10-21T16:39:37.399Z        WARN    openvpn: 'keysize' is used inconsistently, local='keysize 256', remote='keysize 128'
2020-10-21T16:39:37.399Z        INFO    openvpn: [95b1453874f8eee848378fb9f7e7abb3] Peer Connection Initiated with [AF_INET]184.170.242.199:1197
2020-10-21T16:39:38.952Z        INFO    openvpn: TUN/TAP device tun0 opened
2020-10-21T16:39:38.952Z        INFO    openvpn: /sbin/ip link set dev tun0 up mtu 1500
2020-10-21T16:39:38.953Z        INFO    openvpn: /sbin/ip addr add dev tun0 local 10.31.14.6 peer 10.31.14.5
2020-10-21T16:39:38.955Z        INFO    openvpn: UID set to nonrootuser
2020-10-21T16:39:38.955Z        INFO    openvpn: Initialization Sequence Completed
2020-10-21T16:39:38.955Z        INFO    routing: default route found: interface eth0, gateway 172.17.0.1
2020-10-21T16:39:38.955Z        INFO    VPN routing IP address: 184.170.242.199
2020-10-21T16:39:38.956Z        INFO    dns configurator: downloading root hints from https://raw.githubusercontent.com/qdm12/files/master/named.root.updated
2020-10-21T16:39:40.443Z        INFO    dns configurator: downloading root key from https://raw.githubusercontent.com/qdm12/files/master/root.key.updated
2020-10-21T16:39:40.731Z        INFO    dns configurator: generating Unbound configuration
2020-10-21T16:39:41.919Z        INFO    dns configurator: 61358 hostnames blocked overall
2020-10-21T16:39:41.919Z        INFO    dns configurator: 2581 IP addresses blocked overall
2020-10-21T16:39:42.089Z        INFO    dns configurator: starting unbound
2020-10-21T16:39:42.090Z        INFO    dns configurator: using DNS address 127.0.0.1 internally
2020-10-21T16:39:42.090Z        INFO    dns configurator: using DNS address 127.0.0.1 system wide
2020-10-21T16:39:42.346Z        INFO    unbound: init module 0: validator
2020-10-21T16:39:42.346Z        INFO    unbound: init module 1: iterator
2020-10-21T16:39:42.360Z        INFO    unbound: start of service (unbound 1.10.1).
2020-10-21T16:39:42.680Z        INFO    unbound: generate keytag query _ta-4a5c-4f66. NULL IN
2020-10-21T16:39:44.853Z        INFO    dns over tls: DNS over TLS is ready
2020-10-21T16:39:46.012Z        INFO    You are running on the bleeding edge of latest!
2020-10-21T16:39:46.881Z        INFO    ip getter: Public IP address is 184.170.242.199
```

Configuration file:

```
version: "3"
services:
  pia:
    build: https://github.com/qdm12/private-internet-access-docker.git
    image: qmcgaw/private-internet-access:latest
    container_name: pia
    cap_add:
      - NET_ADMIN
    network_mode: bridge
    ports:
      - 8888:8888/tcp #pia
      - 8388:8388/tcp #pia-proxy
      - 8388:8388/udp #pia-proxy
    environment:
      - VPNSP=private internet access old
      - USER=xxxxxx
      - PASSWORD=xxxxxx
      - REGION=US West
      - EXTRA_SUBNETS=192.168.0.0/24
    restart: always
```

Host OS:
synology DSM 6.2.3-25426 Update 2
Also saw it on antiX-17.4

**Comment by dfsimmons7** [comment]
I found the resolution to this issue in https://github.com/qdm12/gluetun/issues/256

I have always used the PPTP/L2TP/SOCKS Username and Password.  With the change to the v4 servers, it looks like they aren't allowing those credentials anymore and you must use your PIA account Username and Password.

**Comment by qdm12** [comment]
Let me reopen the issue. I'll add a log line about this when you use piav4 and get AUTH_FAILED as this seems like a popular issue now! 😄

**Comment by ali5h** [comment]
using PIA username does not fix it for me at least. I am having this problem for the last week

**Comment by qdm12** [comment]
Are you sure you are using the openvpn one? Can you share your logs or at least the error message you get? Thanks!

**Comment by ali5h** [comment]
logs:

```
pia          | 2020-11-15T02:02:35.914Z INFO    OpenVPN version: 2.4.9
pia          | 2020-11-15T02:02:35.915Z INFO    Unbound version: 1.10.1
pia          | 2020-11-15T02:02:35.916Z INFO    IPtables version: v1.8.4
pia          | 2020-11-15T02:02:35.916Z INFO    Settings summary below:
pia          | OpenVPN settings:
pia          | |--User: [redacted]
pia          | |--Password: [redacted]
pia          | |--Verbosity level: 1
pia          | |--Run as root: no
pia          | |--Private Internet Access settings:
pia          |  |--Network protocol: udp
pia          |  |--Regions: us california
pia          |  |--Encryption preset: strong
pia          |  |--Port forwarding: off
pia          | System settings:
pia          | |--User ID: 1000
pia          | |--Group ID: 1000
pia          | |--Timezone:
pia          | |--IP Status filepath: /tmp/gluetun/ip
pia          | DNS over TLS settings:
pia          |  |--DNS over TLS provider:
pia          |   |--cloudflare
pia          |  |--Caching: enabled
pia          |  |--Block malicious: enabled
pia          |  |--Block surveillance: disabled
pia          |  |--Block ads: disabled
pia          |  |--Allowed hostnames:
pia          |   |--
pia          |  |--Private addresses:
pia          |   |--127.0.0.1/8
pia          |   |--10.0.0.0/8
pia          |   |--172.16.0.0/12
pia          |   |--192.168.0.0/16
pia          |   |--169.254.0.0/16
pia          |   |--::1/128
pia          |   |--fc00::/7
pia          |   |--fe80::/10
pia          |   |--::ffff:0:0/96
pia          |  |--Verbosity level: 1/5
pia          |  |--Verbosity details level: 0/4
pia          |  |--Validation log level: 0/2
pia          |  |--IPv6 resolution: disabled
pia          |  |--Update: every 24h0m0s
pia          |  |--Keep nameserver (disabled blocking): no
pia          | Firewall settings:
pia          |  |--VPN input ports:
pia          |  |--Input ports:
pia          |  |--Outbound subnets:
pia          | HTTP Proxy settings: disabled
pia          | ShadowSocks settings: disabled
pia          | HTTP Control server:
pia          |  |--Listening port: 8000
pia          |  |--Logging: true
pia          | Public IP check period: 12h0m0s
pia          | Version information: enabled
pia          | Updater: disabled
pia          |
pia          | 2020-11-15T02:02:35.935Z INFO    storage: Merging by most recent 6734 hardcoded servers and 6734 servers read from /gluetun/servers.json
pia          | 2020-11-15T02:02:35.951Z INFO    routing: default route found: interface eth0, gateway 172.17.0.1
pia          | 2020-11-15T02:02:35.951Z INFO    routing: local subnet found: 172.17.0.0/16
pia          | 2020-11-15T02:02:35.951Z INFO    routing: default route found: interface eth0, gateway 172.17.0.1
pia          | 2020-11-15T02:02:35.951Z INFO    routing: adding route for 0.0.0.0/0
pia          | 2020-11-15T02:02:35.951Z INFO    firewall: firewall disabled, only updating allowed subnets internal list
pia          | 2020-11-15T02:02:35.951Z INFO    routing: default route found: interface eth0, gateway 172.17.0.1
pia          | 2020-11-15T02:02:35.951Z INFO    openvpn configurator: checking for device /dev/net/tun
pia          | 2020-11-15T02:02:35.951Z INFO    firewall: enabling...
pia          | 2020-11-15T02:02:35.958Z INFO    firewall: enabled successfully
pia          | 2020-11-15T02:02:35.958Z INFO    healthcheck: listening on 127.0.0.1:9999
pia          | 2020-11-15T02:02:35.958Z INFO    Launching standard output merger
pia          | 2020-11-15T02:02:35.958Z INFO    http server: listening on 0.0.0.0:8000
pia          | 2020-11-15T02:02:35.958Z INFO    dns over tls: falling back on plaintext DNS at address 1.1.1.1
pia          | 2020-11-15T02:02:35.958Z INFO    dns configurator: using DNS address 1.1.1.1 internally
pia          | 2020-11-15T02:02:35.958Z INFO    dns configurator: using DNS address 1.1.1.1 system wide
pia          | 2020-11-15T02:02:35.958Z INFO    firewall: setting VPN connection through firewall...
pia          | 2020-11-15T02:02:35.959Z INFO    openvpn configurator: starting openvpn
pia          | 2020-11-15T02:02:35.960Z INFO    openvpn: OpenVPN 2.4.9 x86_64-alpine-linux-musl [SSL (OpenSSL)] [LZO] [LZ4] [EPOLL] [MH/PKTINFO] [AEAD] built on Apr 20 2020
pia          | 2020-11-15T02:02:35.960Z INFO    openvpn: library versions: OpenSSL 1.1.1g  21 Apr 2020, LZO 2.10
pia          | 2020-11-15T02:02:35.962Z INFO    openvpn: CRL: loaded 1 CRLs from file [[INLINE]]
pia          | 2020-11-15T02:02:35.962Z INFO    openvpn: TCP/UDP: Preserving recently used remote address: [AF_INET]143.244.49.144:1197
pia          | 2020-11-15T02:02:35.962Z INFO    openvpn: UDP link local: (not bound)
pia          | 2020-11-15T02:02:35.962Z INFO    openvpn: UDP link remote: [AF_INET]143.244.49.144:1197
pia          | 2020-11-15T02:02:36.182Z INFO    openvpn: [losangeles409] Peer Connection Initiated with [AF_INET]143.244.49.144:1197
pia          | 2020-11-15T02:02:37.434Z ERROR   openvpn: AUTH: Received control message: AUTH_FAILED
pia          |
pia          |   (IF YOU ARE USING PIA servers, MAYBE CHECK OUT https://github.com/qdm12/gluetun/issues/265)
pia          |

```

**Comment by ali5h** [comment]
nvm, working now

**Comment by 0xE232FE** [comment]
Thanks for updating the servers.json in container.

But I get the BAD AUTH Error now.

I switched to Nextgen Configs and it doesn't seem to work on all locations.

I am using the Pia Username starting with "p" looking like p1234567 and the normal Pia Password. On some locations I get Bad Auth Error also on my OpenVPN Client on Windows/Linux and Mac. Its an Privateinternetaccess issue. I am actually escalating this at the Support Level.  And I will keep up here with the Result of the Pia Support.



```
=========================================
================ Gluetun ================
=========================================
==== A mix of OpenVPN, DNS over TLS, ====
======= Shadowsocks and HTTP proxy ======
========= all glued up with Go ==========
=========================================
=========== For tunneling to ============
======== your favorite VPN server =======
=========================================
=== Made with ❤️ by github.com/qdm12 ====
=========================================

Running version latest built on 2020-11-14T22:21:52Z (commit 6896cf4)

📣 Support for Privado

🔧 Need help? https://github.com/qdm12/gluetun/issues/new
💻 Email? quentin.mcgaw@gmail.com
☕ Slack? Join from the Slack button on Github
💸 Help me? https://github.com/sponsors/qdm12
2020-11-15T17:20:26.874Z INFO IPtables version: v1.8.4
2020-11-15T17:20:26.877Z INFO OpenVPN version: 2.4.9
2020-11-15T17:20:26.881Z INFO Unbound version: 1.10.1
2020-11-15T17:20:26.882Z INFO Settings summary below:
OpenVPN settings:
|--User: [redacted]
|--Password: [redacted]
|--Verbosity level: 1
|--Run as root: no
|--Private Internet Access settings:
|--Network protocol: udp
|--Regions: de berlin
|--Encryption preset: strong
|--Port forwarding: off
System settings:
|--User ID: 1000
|--Group ID: 1000
|--Timezone:
|--IP Status filepath: /tmp/gluetun/ip
DNS over TLS settings:
|--DNS over TLS provider:
|--cloudflare
|--Caching: enabled
|--Block malicious: enabled
|--Block surveillance: disabled
|--Block ads: disabled
|--Allowed hostnames:
|--
|--Private addresses:
|--127.0.0.1/8
|--10.0.0.0/8
|--172.16.0.0/12
|--192.168.0.0/16
|--169.254.0.0/16
|--::1/128
|--fc00::/7
|--fe80::/10
|--::ffff:0:0/96
|--Verbosity level: 1/5
|--Verbosity details level: 0/4
|--Validation log level: 0/2
|--IPv6 resolution: disabled
|--Update: every 24h0m0s
|--Keep nameserver (disabled blocking): no
Firewall settings:
|--VPN input ports:
|--Input ports:
|--Outbound subnets:
HTTP Proxy settings: disabled
ShadowSocks settings: disabled
HTTP Control server:
|--Listening port: 8000
|--Logging: true
Public IP check period: 12h0m0s
Version information: enabled
Updater: disabled

2020-11-15T17:20:26.985Z INFO storage: Merging by most recent 6734 hardcoded servers and 6733 servers read from /gluetun/servers.json
2020-11-15T17:20:26.986Z INFO storage: Using Private Internet Access servers from file (18h12m51s more recent)
2020-11-15T17:20:27.094Z INFO routing: default route found: interface eth0, gateway 172.17.0.1
2020-11-15T17:20:27.094Z INFO routing: local subnet found: 172.17.0.0/16
2020-11-15T17:20:27.095Z INFO routing: default route found: interface eth0, gateway 172.17.0.1
2020-11-15T17:20:27.096Z INFO routing: adding route for 0.0.0.0/0
2020-11-15T17:20:27.096Z INFO firewall: firewall disabled, only updating allowed subnets internal list
2020-11-15T17:20:27.097Z INFO routing: default route found: interface eth0, gateway 172.17.0.1
2020-11-15T17:20:27.097Z INFO openvpn configurator: checking for device /dev/net/tun
2020-11-15T17:20:27.097Z WARN TUN device is not available: open /dev/net/tun: no such file or directory
2020-11-15T17:20:27.097Z INFO openvpn configurator: creating /dev/net/tun
2020-11-15T17:20:27.098Z INFO firewall: enabling...
2020-11-15T17:20:27.111Z INFO firewall: enabled successfully
2020-11-15T17:20:27.111Z INFO healthcheck: listening on 127.0.0.1:9999
2020-11-15T17:20:27.112Z INFO Launching standard output merger
2020-11-15T17:20:27.113Z INFO firewall: setting VPN connection through firewall...
2020-11-15T17:20:27.114Z INFO dns over tls: falling back on plaintext DNS at address 1.1.1.1
2020-11-15T17:20:27.114Z INFO dns configurator: using DNS address 1.1.1.1 internally
2020-11-15T17:20:27.114Z INFO dns configurator: using DNS address 1.1.1.1 system wide
2020-11-15T17:20:27.114Z INFO http server: listening on 0.0.0.0:8000
2020-11-15T17:20:27.117Z INFO openvpn configurator: starting openvpn
2020-11-15T17:20:27.120Z INFO openvpn: OpenVPN 2.4.9 x86_64-alpine-linux-musl [SSL (OpenSSL)] [LZO] [LZ4] [EPOLL] [MH/PKTINFO] [AEAD] built on Apr 20 2020
2020-11-15T17:20:27.120Z INFO openvpn: library versions: OpenSSL 1.1.1g 21 Apr 2020, LZO 2.10
2020-11-15T17:20:27.122Z INFO openvpn: CRL: loaded 1 CRLs from file [[INLINE]]
2020-11-15T17:20:27.122Z INFO openvpn: TCP/UDP: Preserving recently used remote address: [AF_INET]154.13.1.119:1197
2020-11-15T17:20:27.122Z INFO openvpn: UDP link local: (not bound)
2020-11-15T17:20:27.123Z INFO openvpn: UDP link remote: [AF_INET]154.13.1.119:1197
2020-11-15T17:20:33.373Z INFO openvpn: [berlin423] Peer Connection Initiated with [AF_INET]154.13.1.119:1197
2020-11-15T17:20:34.414Z ERROR openvpn: AUTH: Received control message: AUTH_FAILED

(IF YOU ARE USING PIA servers, MAYBE CHECK OUT https://github.com/qdm12/gluetun/issues/265)

2020-11-15T17:20:34.414Z INFO openvpn: SIGUSR1[soft,auth-failure] received, process restarting
2020-11-15T17:20:44.415Z INFO openvpn: TCP/UDP: Preserving recently used remote address: [AF_INET]154.13.1.119:1197
2020-11-15T17:20:44.415Z INFO openvpn: UDP link local: (not bound)
2020-11-15T17:20:44.416Z INFO openvpn: UDP link remote: [AF_INET]154.13.1.119:1197
2020-11-15T17:20:48.809Z INFO openvpn: [berlin423] Peer Connection Initiated with [AF_INET]154.13.1.119:1197
```

**Comment by npawelek** [comment]
@0xE232FE What did PIA support say? I'm running `latest` and seeing the same issue. I'm using the proper PIA credentials and I even tested disabling 2FA to see if that was causing the issue.

**Comment by npawelek** [comment]
Quick update... I did a bit more testing and switching to a different region solved my `AUTH_FAILED` issue. The region I was using is listed in the nextgen servers list, but wasn't working. Once I switched it seems fine.

**Comment by qdm12** [comment]
Please see #256 

TLDR: It's most likely a PIA issue, and I'm working on compiling openvpn 2.5.0 (newer) to see if that solves it (as only 2.4.9 is available from Alpine).

**Comment by qdm12** [comment]
Just a tiny FYI, `qmcgaw/gluetun` and `qmcgaw/gluetun:v3.12.0` now run with openvpn 2.5.0 😉

**Comment by hugalafutro** [comment]
I use gluetun + pia on 2 machines both same docker-compose.yml, same environment variables, BUT one uses port forwarding. I only see this issue every now and then on the one which uses port forwarding. 

If I happen to change server on that one then rebuild container chances are it will come up with AUTH_FAILED and link to this issue. To fix it, strangely enough changing to previous server, rebuilding (comes up healthy straight away), then changing to the server that bugged out in 1st place, rebuild and it suddenly comes up heatlhy as if nothing ever happened.

**Comment by qdm12** [comment]
I just pushed something to match the settings they have on their openvpn configuration files more closely (removed ping instructions, added `disable-occ`). If it continues failing (repull latest), I guess their server or their configuration files would be at fault 🤔

**Comment by hugalafutro** [comment]
Maybe something on PIA's side timing out or some such as I just had it happen again, one `docker-compose down && docker-compose up -d` later everything ok again. The only change I made this time was I included new port to export for the gluetun service as I added new container into docker-compose.yml.

here's the gluetun part of my stack
```
  gluetun:
    image: qmcgaw/private-internet-access
    container_name: pia_gluetun
    cap_add:
      - NET_ADMIN
    network_mode: bridge
    ports:
      - 9888:8888/tcp # HTTP proxy
      - 8388:8388/tcp # Shadowsocks
      - 8388:8388/udp # Shadowsocks
      - 8000:8000/tcp # Built-in HTTP control server
# other containers ports
      - 8112:8112     # deluge webui
      - 58846:58846   # deluge daemon
      - 6767:6767     # bazarr
      - 8989:8989     # sonarr
      - 7878:7878     # radarr
      - 8686:8686     # lidarr
      - 9117:9117     # jackett
      - 8191:8191     # flaresolverr
      - 8902:8902     # torrents-csv
    volumes:
      - ./data_gluetun:/gluetun
      - ./data_gluetun/port_forward:/tmp/gluetun/forwarded_port
    environment:
      - VPNSP=private internet access
      - TZ=${TIMEZONE}
      - OPENVPN_USER=${PIA_USER}
      - OPENVPN_PASSWORD=${PIA_PASS}
      - REGION=${PIA_REGION}
      - PORT_FORWARDING=on
      - FIREWALL_OUTBOUND_SUBNETS=192.168.1.0/24
      - HTTPPROXY=on
#      - HTTPPROXY_USER=${PROXY_USER}
#      - HTTPPROXY_PASSWORD=${PROXY_PASS}
      - SHADOWSOCKS=on
      - SHADOWSOCKS_PASSWORD=${SHADOW_PASS}
    restart: unless-stopped
```

**Comment by qdm12** [comment]
@hugalafutro how did you update the settings? With `docker-compose up -d`?

**Comment by hugalafutro** [comment]
yes, I use following shell script to update all the containers:
```
#!/bin/sh
docker-compose pull
docker-compose build deluge
docker-compose down
docker-compose up -d
docker image prune -f
docker volume prune -f
```
which is what I've ran after updating the compose with the new port range

**Comment by qdm12** [comment]
I'm not sure, I'd need to your full logs when the issue happens. It could be an issue on PIA's side too as it happened before. Please create another issue or discussion on Github with your logs, I'll lock this thread now as it's getting quite long these days.
